next-docs/pages/guides/hello-world/add-table.mdx

```diff
@@ -0,0 +1,340 @@
+Click the **Store data** tab and select the table **:History**.
+
+However, you can also add the history to the user interface.
+The directions here apply to the `vanilla` client template, if you use anything else you'll need to modify them as appropriate.
+
+1. Edit `packages/client/src/index.ts`.
+
+   - Import two more functions we need.
+   - In `components.Counter.update$.subscribe`, add code that updates the selection of history values.
+   - Add a function that is called when the user selects a different option.
+
+   <CollapseCode>
+
+   ```ts filename="index.ts" copy showLineNumbers {3-4,18-22,32-38}
+   import { setup } from "./mud/setup";
+   import mudConfig from "contracts/mud.config";
+   import { encodeEntity } from "@latticexyz/store-sync/recs";
+   import { getComponentValueStrict } from "@latticexyz/recs";
+
+   const {
+     components,
+     systemCalls: { increment },
+     network,
+   } = await setup();
+
+   // Components expose a stream that triggers when the component is updated.
+   components.Counter.update$.subscribe((update) => {
+     const [nextValue, prevValue] = update.value;
+     console.log("Counter updated", update, { nextValue, prevValue });
+     document.getElementById("counter")!.innerHTML = String(nextValue?.value ?? "unset");
+
+     let options: String = "";
+     for (let i = 1; i <= nextValue?.value; i++) {
+       options += `<option value="${i}">${i}</option>`;
+     }
+     document.getElementById("historyValue")!.innerHTML = options;
+   });
+
+   // Just for demonstration purposes: we create a global function that can be
+   // called to invoke the Increment system contract via the world.
+   // (See IncrementSystem.sol)
+   (window as any).increment = async () => {
+     console.log("new counter value:", await increment());
+   };
+
+   (window as any).readHistory = async (counterValue) => {
+     const History = components.History;
+     const entity = encodeEntity(History.metadata.keySchema, { counterValue });
+     const { blockNumber, time } = getComponentValueStrict(History, entity);
+     document.getElementById("blockNumber")!.innerHTML = blockNumber;
+     document.getElementById("timeStamp")!.innerHTML = new Date(parseInt(Number(time) * 1000));
+   };
+
+   if (import.meta.env.DEV) {
+     const { mount: mountDevTools } = await import("@latticexyz/dev-tools");
+     mountDevTools({
+       config: mudConfig,
+       publicClient: network.publicClient,
+       walletClient: network.walletClient,
+       latestBlock$: network.latestBlock$,
+       blockStorageOperations$: network.blockStorageOperations$,
+       worldAddress: network.worldContract.address,
+       worldAbi: network.worldContract.abi,
+       write$: network.write$,
+       recsWorld: network.world,
+     });
+   }
+   ```
+
+   </CollapseCode>
+
+<details>
+<summary>Explanation</summary>
+
+```typescript
+let options: String = "";
```

Comment:
```suggestion
let options: string = "";
```

nit, let's use the primitive type over the class (see https://www.geeksforgeeks.org/what-is-the-difference-between-string-and-string-in-typescript)

Reply:
in fact we can remove `: string` since we're initializing it in the same line, so typescript can infer the type